### PR TITLE
Jenkins migration Phase 1

### PR DESCRIPTION
**Pull request by CyanDevs**

### CODE REVIEW

**Review by radhikaj — APPROVED** [review]

**Comment by radhikaj** [comment]
bors try

File: .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile [hunk 1]
```diff
@@ -248,77 +279,79 @@ def buildWindowsManagedImage(String os_series, String img_name_suffix, String la
...
+                    sudo apt-get -y install ca-certificates curl apt-transport-https lsb-release gnupg
+                    curl -sL https://packages.microsoft.com/keys/microsoft.asc |
+                        gpg --dearmor |
+                        sudo tee /etc/apt/trusted.gpg.d/microsoft.gpg > /dev/null
+                    AZ_REPO=\$(lsb_release -cs)
+                    echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ \$AZ_REPO main" |
+                        sudo tee /etc/apt/sources.list.d/azure-cli.list
+                    ${helpers.WaitForAptLock()}
+                    sudo apt-get update
+                    sudo apt-get -y install azure-cli jq
```
Comment (radhikaj) [inline .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile:328]: what is the jq at the end of this?

File: .jenkins/infrastructure/docker/build_linux_docker_images.Jenkinsfile [hunk 2]
```diff
@@ -19,13 +19,11 @@ pipeline {
         string(name: "INTERNAL_REPO", defaultValue: "https://oejenkinscidockerregistry.azurecr.io", description: "Url for internal Docker repository")
         string(name: "OECI_LIB_VERSION", defaultValue: 'master', description: 'Version of OE Libraries to use')
         string(name: "DEVKITS_URI", defaultValue: 'https://oejenkins.blob.core.windows.net/oejenkins/OE-CI-devkits-d1634ce8.tar.gz', description: "Uri for downloading the OECI Devkit")
-        booleanParam(name: "PUBLISH_DOCKER_HUB", defaultValue: false, description: "Publish container to OECITeam Docker Hub?")
```
Comment (radhikaj) [inline .jenkins/infrastructure/docker/build_linux_docker_images.Jenkinsfile:22]: Do we want this to be removed?

File: .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile [hunk 3]
```diff
@@ -94,78 +94,119 @@ def buildLinuxManagedImage(String os_type, String version, String managed_image_
...
-                        PRIVATE_IP=\$(echo \$VM_DETAILS | jq -r '.privateIps')
-                        rm -f ${WORKSPACE}/scripts/ansible/inventory/hosts-${launch_configuration}
-                        echo "[windows-agents]" >> ${WORKSPACE}/scripts/ansible/inventory/hosts-${launch_configuration}
-                        echo "\$PRIVATE_IP" >> ${WORKSPACE}/scripts/ansible/inventory/hosts-${launch_configuration}
-                        echo "ansible_user: ${JENKINS_USER_NAME}" >> ${WORKSPACE}/scripts/ansible/inventory/host_vars/\$PRIVATE_IP
-                        echo "ansible_password: ${JENKINS_USER_PASSWORD}" >> ${WORKSPACE}/scripts/ansible/inventory/host_vars/\$PRIVATE_IP
-                        echo "ansible_winrm_transport: ntlm" >> ${WORKSPACE}/scripts/ansible/inventory/host_vars/\$PRIVATE_IP
-                        echo "launch_configuration: ${launch_configuration}" >> ${WORKSPACE}/scripts/ansible/inventory/host_vars/\$PRIVATE_IP
+            stage("Deploy VM") {
+                checkout([$class: 'GitSCM',
```
Comment (radhikaj) [inline .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile:164]: In what cases do you checkout GitSCM?

File: .jenkins/infrastructure/docker/build_linux_docker_images.Jenkinsfile (hunk 2 above)
Comment (CyanDevs) [reply .jenkins/infrastructure/docker/build_linux_docker_images.Jenkinsfile:22]: Yes, we shouldn't be publishing to Docker Hub anymore as we don't have an enterprise account under their new usage policies.

File: .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile (hunk 3 above)
Comment (CyanDevs) [reply .jenkins/infrastructure/build_azure_managed_images.Jenkinsfile:164]: I'm not sure I understand this question. We checkout the openenclave repo when the files are necessary. In this case we need the Ansible files to that we can run Ansible to set up the VM.

**Comment by radhikaj** [comment]
bors try

**Comment by CyanDevs** [comment]
bors try-
This is to free up bors for other PRs. Bors wont test anything new for this PR since this is all infrastructure and library changes.

**Comment by CyanDevs** [comment]
bors try
Added in changes to `.jenkins/Jenkinsfile` so now there is something to test :)

**Comment by CyanDevs** [comment]
bors try

File: .jenkins/library/vars/tests.groovy [hunk 4]
```diff
@@ -622,7 +626,7 @@ def checkDevFlows(String version) {
 
 def checkCI() {
     stage('Check CI') {
-        node(globalvars.AGENTS_LABELS["ubuntu-nonsgx"]) {
+        node(globalvars.AGENTS_LABELS["ubuntu-nonsgx-18.04"]) {
```
Comment (rs--) [inline .jenkins/library/vars/tests.groovy:629]: Should this default to 20.04, since 18.04 may be dropped sooner than 20.04.

**Review by rs-- — APPROVED** [review]

File: .jenkins/infrastructure/docker/build_linux_docker_images.Jenkinsfile (hunk 2 above)
Comment (CyanDevs) [reply .jenkins/infrastructure/docker/build_linux_docker_images.Jenkinsfile]: Reverted this removal until I can be sure

**Pull request merged**